Framework/TestHelpers/inc/MantidTestHelpers/ParallelAlgorithmCreation.h

```diff
@@ -0,0 +1,45 @@
+#ifndef PARALLELALGORITHMCREATION_H_
+#define PARALLELALGORITHMCREATION_H_
+
+#include "MantidAPI/IWorkspaceProperty.h"
+#include "MantidKernel/make_unique.h"
+
+namespace Mantid {
+namespace Parallel {
+class Communicator;
+}
+}
+
+namespace ParallelTestHelpers {
+
+/** Create an initialized algorithm that does not store workspaces in the ADS.
+ *
+ * This function is the recommended way to create algorithm for use with
+ * ParallelRunner since it avoids name clashes in the ADS from different ranks.
+ * This factory function is intended for unit tests that need to call algorithms
+ * in MPI tests. The communicator argument is obtained from
+ * ParallelTestHelpers::runParallel(). The algorithm is setup such that
+ * workspaces will not be stored in the ADS, i.e., it is not necessary to set
+ * OutputWorkspace properties, just like for child algorithms. */
+template <class T>
+std::unique_ptr<T> create(const Mantid::Parallel::Communicator &comm) {
+  using namespace Mantid;
+  auto alg = Kernel::make_unique<T>();
+  alg->setChild(true);
+  alg->setCommunicator(comm);
+  alg->initialize();
+  const auto &props = alg->getProperties();
+  for (auto prop : props) {
+    auto wsProp = dynamic_cast<API::IWorkspaceProperty *>(prop);
```

Comment:
You could eliminate the named `wsProp` variable here. But I think it does help readability.